dotnet/src/SemanticKernel.Abstractions/Functions/KernelArguments.cs

```diff
@@ -40,7 +44,7 @@ public KernelArguments(PromptExecutionSettings? executionSettings = null)
     /// If <paramref name="executionSettings"/> is non-null, it is used as the <see cref="ExecutionSettings"/> for this new instance.
     /// Otherwise, if the source is a <see cref="KernelArguments"/>, its <see cref="ExecutionSettings"/> are used.
     /// </remarks>
-    public KernelArguments(IDictionary<string, object?> source, PromptExecutionSettings? executionSettings = null)
+    public KernelArguments(IDictionary<string, object?> source, Dictionary<string, PromptExecutionSettings>? executionSettings = null)
```

Comment:
Proposal: Maybe we should keep `KernelArguments` constructor that accepts `PromptExecutionSettings? executionSettings = null` for easier usability, if we want to pass just default settings. 

We can add new constructor that accepts `Dictionary<string, PromptExecutionSettings>? executionSettings` for more advanced scenarios.

Reply:
Yes, we have that on line 28